cluster-autoscaler/estimator/threshold_based_limiter.go

```diff
@@ -56,9 +60,24 @@ func (tbel *thresholdBasedEstimationLimiter) PermissionToAddNode() bool {
 // after either a node count- of time-based threshold is reached. This is meant to prevent cases
 // where binpacking of hundreds or thousands of nodes takes extremely long time rendering CA
 // incredibly slow or even completely crashing it.
-func NewThresholdBasedEstimationLimiter(maxNodes int, maxDuration time.Duration) EstimationLimiter {
+func NewThresholdBasedEstimationLimiter(limits []BinpackingLimit, maxDuration time.Duration) EstimationLimiter {
+	maxNodes := computeLimit(limits, 0)
 	return &thresholdBasedEstimationLimiter{
-		maxNodes:    maxNodes,
-		maxDuration: maxDuration,
+		maxNodes:        maxNodes,
+		maxNodesDefault: maxNodes,
+		maxDuration:     maxDuration,
+	}
+}
+
+func computeLimit(binpackingLimits []BinpackingLimit, currentLimit int) int {
+	if binpackingLimits == nil || len(binpackingLimits) == 0 {
```

Comment:
nit: This if is not necessary, if the slice is empty then we will just return currentLimit. Fortunately golang handles slices being `nil` gracefully when iterating over them.

Reply:
Good for me to know now :-) Thank you

Reply:
Removed in 7654f75852e2f3bbc63be0a199f53faa38bc3895